src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployAllTask.java

```diff
@@ -36,17 +39,29 @@ public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFact
     this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
   }
 
+  public void setStageDirectory(File stageDirectory) {
+    this.stageDirectory = stageDirectory;
+  }
+
   /** Task Entrypoint : Deploys the app and all of its config files. */
   @TaskAction
   public void deployAllAction() throws AppEngineException {
-    if (!deployConfig.getDeployables().isEmpty()) {
-      getLogger().warn("appengineDeployAll: Ignoring configured deployables.");
-      deployConfig.getDeployables().clear();
+    List<File> deployables = deployConfig.getDeployables();
```

Comment:
Would it make sense to instead just call `deployConfig.setDeployables` at the bottom?

Reply:
Looks like my assertions fail when I do this. I think it's because deployConfig is a mock.

Reply:
Ah okay. There's the [`ArgumentCaptor`](https://static.javadoc.io/org.mockito/mockito-core/2.6.9/org/mockito/ArgumentCaptor.html) thing you could use to capture the argument passed into a mocked method and assert things on that. Otherwise, I'd say probably just change L69 to use `deployables` as well just to keep things consistent.

Reply:
Oh wait, I was looking at the wrong thing. Yeah I think this should work.